src/SonarQube.Client/Api/V7_20/GetIssuesWithComponentRequest.cs

```diff
@@ -0,0 +1,56 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Newtonsoft.Json;
+
+namespace SonarQube.Client.Api.V7_20
+{
+    internal class GetIssuesWithComponentRequest : GetIssuesRequest
+    {
+        // this class should not be used on it's own
+        private protected GetIssuesWithComponentRequest(){}
```

Comment:
I think abstract class with abstract property would be slightly better solution here. You'd get rid of the artificial constructor.

Reply:
I think one of the tools (don't remember which one 😁 ) suggested me to change from abstract to concrete class with protected constructor. It seemed like a good idea to me, but maybe the tool was too afraid of abstract classes

Reply:
It is our rule. It raises when there's no abstract member on the class. The sole purpose of this one is to override the property, so having the property abstract to enforce the action would not make it raise again